addons/mail/models/res_users_settings.py

```diff
@@ -16,6 +26,32 @@ class ResUsersSettings(models.Model):
     voice_active_duration = fields.Integer(string="Duration of voice activity in ms", help="How long the audio broadcast will remain active after passing the volume threshold")
     volume_settings_ids = fields.One2many('res.users.settings.volumes', 'user_setting_id', string="Volumes of other partners")
 
+    @api.constrains('notification_preference')
```

Comment:
Missing triggers on the constraint; notably I bet it won't be fired if you change user's status.